midend/interpreter.cpp

```diff
@@ -310,33 +310,45 @@ void SymbolicStruct::dbprint(std::ostream& out) const {
 }
 
 SymbolicHeaderUnion::SymbolicHeaderUnion(const IR::Type_HeaderUnion* type,
-                               bool uninitialized,
-                               const SymbolicValueFactory* factory) :
-        SymbolicStruct(type, uninitialized, factory),
-        valid(new SymbolicBool(false)) {}
-
-void SymbolicHeaderUnion::setValid(bool v) {
-    if (!v)
-        setAllUnknown();
-    valid = new SymbolicBool(v);
+                                         bool uninitialized,
+                                         const SymbolicValueFactory* factory) :
+        SymbolicStruct(type, uninitialized, factory) {}
+
+SymbolicBool* SymbolicHeaderUnion::isValid() const {
+    int validFields = 0;
+    for (auto f : type->to<IR::Type_StructLike>()->fields) {
+         if (fieldValue.count(f->name.name)) {
+             auto fieldValid = fieldValue.at(f->name.name)->checkedTo<SymbolicHeader>()->valid;
+             if (!fieldValid->isKnown() || fieldValid->isUninitialized()) {
+                return new SymbolicBool(false);
```

Comment:
if any field has the value "unknown" or "uninitialized", the result is "unknown"

Reply:
Oh I got the error

Reply:
I do not quite understand how we can return "unknown", do something similar to this?
`return new SymbolicBool(ScalarValue::ValueState::NotConstant))`